elrond-wasm/src/contract_base/wrappers/serializer.rs

```diff
@@ -54,6 +54,18 @@ where
         value
     }
 
+    pub fn top_decode_from_managed_buffer_custom_message<T: TopDecode>(
```

Comment:
Make `top_decode_from_managed_buffer` call this, there is a little duplicate code there.

Also make it take a &'static str, I think it would be nice to move in that direction. It makes code nicer to read and to debug.